plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerExtensionService.java

```diff
@@ -477,46 +508,42 @@ private void fixLocation(ExtendedSymbolInformation symbol) {
-            .executeCommand(
-                new ExecuteCommandParams(
-                    Commands.IDENTIFY_FQN_IN_RESOURCE,
-                    ImmutableList.of(prefixURI(filePath), String.valueOf(lineNumber))));
-
-    try {
-      return (String) result.get(10, TimeUnit.SECONDS);
-    } catch (JsonSyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
-      throw new JsonRpcException(-27000, e.getMessage());
-    }
+    Type type = new TypeToken<String>() {}.getType();
+    return doGetOne(
+        Commands.IDENTIFY_FQN_IN_RESOURCE,
+        ImmutableList.of(prefixURI(filePath), String.valueOf(lineNumber)),
+        type);
   }
 
   public Location findResourcesByFqn(String fqn, int lineNumber) {
-    CompletableFuture<Object> result =
-        getLanguageServer()
-            .getWorkspaceService()
-            .executeCommand(
-                new ExecuteCommandParams(
-                    Commands.FIND_RESOURCES_BY_FQN,
-                    ImmutableList.of(fqn, String.valueOf(lineNumber))));
+    Type type = new TypeToken<List<Either<String, ResourceLocation>>>() {}.getType();
+    List<Either<String, ResourceLocation>> location =
+        doGetList(
+            Commands.FIND_RESOURCES_BY_FQN,
+            ImmutableList.of(fqn, String.valueOf(lineNumber)),
+            type);
+
+    Either<String, ResourceLocation> l = location.get(0);
+
+    if (l.isLeft()) {
+      return new LocationImpl(l.getLeft(), lineNumber, null);
+    } else {
+      return new LocationImpl(
+          l.getRight().getFqn(), lineNumber, true, l.getRight().getLibId(), null);
+    }
+  }
 
-    try {
-      List<Either<String, ResourceLocation>> location =
-          gson.fromJson(
-              gson.toJson(result.get(10, TimeUnit.SECONDS)),
-              new com.google.common.reflect.TypeToken<
-                  List<Either<String, ResourceLocation>>>() {}.getType());
-      Either<String, ResourceLocation> l = location.get(0);
-
-      if (l.isLeft()) {
-        return new LocationImpl(l.getLeft(), lineNumber, null);
-      } else {
-        return new LocationImpl(
-            l.getRight().getFqn(), lineNumber, true, l.getRight().getLibId(), null);
+  public JobResult updateWorkspace(UpdateWorkspaceParameters updateWorkspaceParameters) {
+    if (updateWorkspaceParameters.getAddedProjectsUri().isEmpty()) {
+      if (!findInitializedLanguageServer().isPresent()) {
+        JobResult jobResult = new JobResult();
+        jobResult.setMessage(
+            "Skipped. Language server not initialized. Workspace updating is not required.");
+        return jobResult;
       }
-    } catch (JsonSyntaxException | InterruptedException | ExecutionException | TimeoutException e) {
-      throw new JsonRpcException(-27000, e.getMessage());
     }
+
+    Type type = new TypeToken<JobResult>() {}.getType();
+    return doGetOne(Commands.UPDATE_WORKSPACE, singletonList(updateWorkspaceParameters), type, 60);
```

Comment:
We have some constant about timeout here `org.eclipse.che.ide.ext.java.shared.Constants`, maybe move this number or change existed. What do you think?